### PR TITLE
7038 90.1 metric

### PR DESCRIPTION
**Pull request by TMaile**
Pull request overview
---------------------
 - Fixes #ISSUENUMBERHERE (IF THIS IS A DEFECT)
 - DESCRIBE PURPOSE OF THIS PULL REQUEST

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by JasonGlazer** [comment]
I found a nice presentation on the differences between the SEER and HSPF and SEER2 and HSPF2

https://www.ahrinet.org/Portals/AHRI%20CAC%20HP%20Appendix%20M1%20Presentation%20-%20AHR%20Expo.pdf

If you need a contact person at AHRI, let me know.

**Comment by nealkruis** [comment]
@TMaile I've been working on managing the transition to SEER2/HSPF2 for RESNET standards. Outside of the changes to how the heating load bins are calculated, the biggest difference is the static pressure used to determine fan blower efficacy. Since EnergyPlus doesn't actually characterize fan performance at different external static pressures, I'm not sure if it will be possible to capture this impact from the information available in EnergyPlus. Maybe the EnergyPlus calculations are based on the default coil-only values for fan efficacy and flow rate?

**Comment by mjwitte** [comment]
@nealkruis Two of the EnergyPlus coils appear to have a user input for this (but with no default). Coil:Cooling:DX:CurveFit:Performance and Coil:Cooling:DX:TwoSpeed have this field. I don't know what the other DX coils assume.
```
   N4, \field Unit Internal Static Air Pressure
       \note Enter pressure drop for the unit containing the coil.
       \note This value is only used to calculate Energy Efficiency Ratio
       \note (EER), Integrated Energy Efficiency Ratio (IEER), and the
       \note Standard Rating (Net) Cooling Capacity. This value is not
       \note used for modeling the evaporator fan during simulations.
       \type real
       \units Pa
       \minimum> 0.0
```

**Comment by nealkruis** [comment]
I haven't looked at how this input is used, but for AHRI 210/240 calculations it is determined by the standard, not the equipment. Since this references IEER, I'm guessing it's not relevant to 210/240 scope equipment.

**Comment by Nigusse** [comment]
The single speed and multi speed DX coils have an input field `Rated Evaporator Fan Power Per Volume Flow Rate` to account for `static pressure rise` and `fan total efficiency` combined (= `DeltaP` / `fan efficiency`) which is meant as a representative value for the AHRI test condition. This value is used to calculate the fan power at the AHRI rated condition and net cooling capacity. Note that the AHRI test conditions static pressure seen by the fan can be (or is) different from the supply fan pressure rise at design or installed condition. It is for this reason that we added this input field in EnergyPlus and used for standard ratings calculation only. See below the input field description. 

```
  N5 , \field Rated Evaporator Fan Power Per Volume Flow Rate
       \note Enter the evaporator fan power per air volume flow rate at the rated test conditions.
       \note The test conditions vary external static pressure based on total cooling capacity.
       \note This value is only used to calculate Seasonal Energy Efficiency Ratio (SEER), Energy
       \note Efficiency Ratio (EER), Integrated Energy Efficiency Ratio (IEER), and the Standard Rating
       \note (Net) Cooling Capacity which will be outputs in the EnergyPlus eio file. This value is not
       \note used for modeling the evaporator fan during simulations.
```

**Comment by nealkruis** [comment]
I think this means that we don't need any code changes for SEER2. The user needs to input this value corresponding to the 2023 standard external static pressure. I'm not sure where they are going to get that information, though. Manufacturers don't provide it and, based on our conversations with them in SPC 205, they really don't want to share this value.

**Comment by RichSee** [comment]
@nealkruis, @JasonGlazer , @mjwitte , @Nigusse -- seems like it would be a good idea to schedule a call to discuss.  Can you tell me about your availabilities tomorrow and Friday?  Tobias is out on Thursday.  Mornings are best  ;-)   Thanks!

**Comment by nealkruis** [comment]
Could we tack it on to the Team call tomorrow? It's a good topic for a technicalities call too, but I suspect next week's call will be canceled because of the ASHRAE conference. Some of us might be travel on Friday already.

**Comment by Nigusse** [comment]
I am available tomorrow but not on Friday.

**Comment by RichSee** [comment]
Sounds like a good idea Neal.  We look forward to it.

**Comment by TMaile** [comment]
Thanks @amirroth for your comments. We did push this code a bit early, anyway we fixed the issues, except the too many arguments. We will be looking in creating a struct once we have all variations of SEER2 implemented.

**Comment by RichSee** [comment]
@Myoldmopar, @Nigusse , @JasonGlazer , @nealkruis  
Hello all,
There were a lot of changes between the 2017 and 2023 of the ANSI/AHRI 210-240 Standard for calculating metrics for DX Cooling and DX Heating coils.  Changes were predominantly for the DX Heating coil metrics.
Consequently, there is a fair amount of new code.
Previous metric calculation (for the 2017 version of the standard) have been maintained and re-verified. 
Both sets of metrics (2017 & 2023) are now reported in both the eio file and the tabular HTML report.
We have updated existing unit tests and added new unit tests as required.
Documentation has also been updated -- in I/O Reference and Engineering Reference.
Please send us your comments, questions, and concerns.
@Brijendra21, @TMaile, @RichSee

**Comment by Myoldmopar** [comment]
My todo:
 - [x] Looks like datasets are not being scanned for tabs, add that to the check
 - [x] Apply formatting locally
 - [x] Check why build is failing
 - [x] Run script to update objects, check for regressions
 - [x] Add the new field to ExpandObjects
 - [x] Manually fix any straggling unit test text updates
 - [ ] Add Fortran rules implementation

**Comment by Myoldmopar** [comment]
I think the Windows build failure is actually an anomaly.  I'll investigate only if it happens again.  I've locally got the formatting fixed, and confirmed the test suite status (all the IDFs missing the new fields are failing).

**Comment by Myoldmopar** [comment]
@Brijendra21 I pushed a change that fixes up almost all the issues.  The main piece missing is writing the Fortran transition implementation itself, which I will do tomorrow morning.  Please pull down the latest version of this branch before making any more changes.  And one thing you could do to help me out is update the `src/Transtiion/InputRules/Rules22-1-0-to-22-2-0.md` file with a description of the `Coil:Heating:DX:MultiSpeed` changes.  That way I make sure I have the field numbers correct when I write the Fortran implementation.

**Comment by RichSee** [comment]
@Myoldmopar - I made the changes to the transition rules file and checked in.  Brijendra also made several updates.
all were pushed and the pipeline checks passed.  Please let us know if we missed anything  ;-)
I will be on the Iteration call in 15 minutes.

File: src/ExpandObjects/epfilter.f90 [hunk 1]
```diff
@@ -17430,6 +17438,7 @@ SUBROUTINE CreateNewUnitary
     CALL AddToObjFld('Speed 1 Gross Rated Heating COP {W/W}', base + ussHPHeatCoilCOPOff,' ')
     CALL AddToObjStr('Speed 1 Rated Air Flow Rate {m3/s}','autosize')
     CALL AddToObjStr('Speed 1 Rated Supply Air Fan Power Per Volume Flow Rate {W/(m3/s)}','')
+    CALL AddToObjStr('Speed 1 2023 Rated Evaporator Fan Power Per Volume Flow Rate {W/(m3/s)}','')
```
Comment (mjwitte) [inline src/ExpandObjects/epfilter.f90]: And for heating coils "Rated Supply Air Fan Power . . ." (not Evaporator). And it looks like some dx heating coils may have been missed?

File: src/ExpandObjects/epfilter.f90 [hunk 2]
```diff
@@ -14227,6 +14227,7 @@ SUBROUTINE CreateNewUnitary
     CALL AddToObjFld('Rated COP', base + usCoolCoilCOPOff,' ')
     CALL AddToObjStr('Rated Air Flow Rate {m3/s}', 'autosize')
     CALL AddToObjStr('Rated Evaporator Fan Power per Volume Flow Rate {W/(m3/s)}','')
+    CALL AddToObjStr('2023 Rated Evaporator Fan Power Per Volume Flow Rate {W/(m3/s)}','')
```
Comment (mjwitte) [inline src/ExpandObjects/epfilter.f90]: For cooling and heating the existing field names should be "2017 Rated ..." if I'm following correctly.

**Review by mjwitte — COMMENTED** [review]
@Myoldmopar

**Comment by Myoldmopar** [comment]
OK.  I fixed the ExpandObjects code so that it will properly change the field name of the 2017 rated value.  Everything seems good there.

And now the transition rules are implemented.  I converted 4 different 22.1 files with combinations of the 5 coils touched here.  They transitioned properly and ran fine after transition to 22.2.  I will check whether this needs anything else, but I think it's either done or close to done.

File: src/ExpandObjects/epfilter.f90 (hunk 1 above)
Comment (Myoldmopar) [reply src/ExpandObjects/epfilter.f90]: I think it's good to go now.  I believe it is working properly in all the instances of `CALL CreateNewObj('CoilTypeHere')`

File: src/ExpandObjects/epfilter.f90 (hunk 2 above)
Comment (Myoldmopar) [reply src/ExpandObjects/epfilter.f90]: Yep fixed it to use the new 2017 field name.  Tried to be very consistent with the exact field names in the transition rules description file.

**Comment by Myoldmopar** [comment]
I realized an integration test file and a unit test were both failing as the inputs were missing the new field.  I updated them and the full integration and unit test suites pass now.  Custom check and format checks all pass locally.  As for regressions, they are generally clean, but the EIO output is a little funny.  Now there are two rows that say similar things, with no real differentiation:

```
  DX Cooling Coil Standard Rating Information, Coil:Cooling:DX:SingleSpeed, WEST DX COOLING COIL, 127600.6, 2.66, 9.06, 9.54, 9.02, 9.44
+ DX Cooling Coil Standard Rating Information, Coil:Cooling:DX:SingleSpeed, WEST DX COOLING COIL, 126745.6, 2.59, 8.84, 9.30, 9.05,  
```

As much as I'd like it fixed, I think this will have to wait until after IO freeze.  I plan to merge this shortly.  If anyone would like to see anything else before merge, let me know soon.

**Comment by Myoldmopar** [comment]
OK everything seems to be in shape, except there are 2 unused variable build warnings.  I cleaned those and verified format/custom check still passes fine.  I'm pushing that change and then merging so we don't waste CI cycles on it.  Thanks @TMaile @RichSee and @mjwitte for reviewing.

**Pull request merged**